FestivalPlanner/screenings/views.py

```diff
@@ -178,3 +250,83 @@ def form_valid(self, form):
     def get_success_url(self):
         return reverse('screenings:day_schema')
 
+
+class ScreeningDetailView(LoginRequiredMixin, SingleObjectMixin, FormView):
+    model = Screening
+    form_class = AttendanceForm
+    template_name = 'screenings/details.html'
+    http_method_names = ['get', 'post']
+    fan_action = FanAction('update')
+    errors_cookie = Cookie('errors', [])
+    update_by_attends = {True: 'joins', False: "couldn't come"}
+    fans = FilmFan.film_fans.all()
+    object = None
+    screening = None
+    initial_attendance_by_fan = None
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        session = self.request.session
+        initialize_log(request.session, 'Update attendances')
+        _ = self.errors_cookie.get(session)
+        self.screening = self.get_object()
+        self.fan_action.init_action(session, screening=self.screening)
+        manager = Attendance.attendances
+        self.initial_attendance_by_fan = {f: bool(manager.filter(screening=self.screening, fan=f))for f in self.fans}
+
+    def get_context_data(self, **kwargs):
+        super_context = super().get_context_data(**kwargs)
+        session = self.request.session
+        duration = self.screening.end_dt - self.screening.start_dt
+        fan_specs = []
+        for fan in self.fans:
+            attends = self.initial_attendance_by_fan[fan]
+            fan_specs.append({
+                'fan': fan.name,
+                'attends': attends,
+                'checked': attends,
+            })
+        new_context = {
+            'title': 'Screening Details',
+            'subtitle': f'Screening of {self.screening.film}',
+            'screening': self.screening,
+            'duration': duration,
+            'minutes': f'{duration.total_seconds() / 60:.0f}{MINUTES_STR}',
+            'film_description': ResultsView.get_description(self.screening.film),
+            'fan_specs': fan_specs,
+            'form_errors': self.errors_cookie.get(session),
+        }
+        self.errors_cookie.remove(session)
+        context = add_base_context(self.request, super_context | new_context)
+        return context
+
+    def form_valid(self, form):
+        new_attendance_by_fan = {fan: fan.name in self.request.POST for fan in self.fans}
+        self._update_attendance(new_attendance_by_fan)
+        return super().form_valid(form)
+
+    def form_invalid(self, form):
+        errors = self.errors_cookie.get(self.request.session)
+        errors.extend(wrap_up_form_errors(form.errors))
+        self.errors_cookie.set(self.request.session, errors)
+        super().form_invalid(form)
+        return self._dirty_response()
```

Comment:
Please consider to drop `form_invalid()` and all supporting code.
The form can never be invalid.

Reply:
Done.